pkg/isb/stores/redis/read.go

```diff
@@ -55,33 +48,73 @@ type BufferReadInfo struct {
 var _ isb.BufferReader = (*BufferRead)(nil)
 
 // NewBufferRead returns a new redis buffer reader.
-func NewBufferRead(ctx context.Context, client *redisclient.RedisClient, name string, group string, consumer string, opts ...Option) isb.BufferReader {
-	options := &options{
-		infoRefreshInterval: time.Second,
-		readTimeOut:         time.Second,
-		checkBackLog:        true,
+func NewBufferRead(ctx context.Context, client *redisclient.RedisClient, name string, group string, consumer string, opts ...redisclient.Option) isb.BufferReader {
+	options := &redisclient.Options{
+		InfoRefreshInterval: time.Second,
+		ReadTimeOut:         time.Second,
+		CheckBackLog:        true,
 	}
 
 	for _, o := range opts {
-		o.apply(options)
+		o.Apply(options)
 	}
```

Comment:
needed to capitalize these in order to export them, since they're now in a shared directory